### ISSUE TITLE
IBM Aspera Connect cannot be uninstalled: No such file or directory

### ISSUE THREAD

**Issue by chohner** [Homebrew/homebrew-cask#99043]
#### General troubleshooting steps

- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [x] I have retried my command with `--force`.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

#### Description of issue

`brew uninstall ibm-aspera-connect` fails with:
```
Error: Failure while executing; `\~/Library/Application\ Support/Aspera/Aspera\ Connect/uninstall_connect.sh -f` exited with 127. Here's the output:
No such file or directory - ~/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh
```

The correct path to the uninstall script seems to match what is shown (except the leading backlash `\`):
```
~/Library/Application\ Support/Aspera/Aspera\ Connect/uninstall_connect.sh
```

#### Command that failed

```
~ ❯ brew uninstall ibm-aspera-connect

==> Uninstalling Cask ibm-aspera-connect
==> Running uninstall script ~/Library/Application Support/Aspera/Aspera Connect
Error: Failure while executing; `\~/Library/Application\ Support/Aspera/Aspera\ Connect/uninstall_connect.sh -f` exited with 127. Here's the output:
No such file or directory - ~/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh
```

#### Output of command with `--verbose --debug`

```
~ ❯ brew uninstall ibm-aspera-connect --verbose --debug
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/lame.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gmp.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cppcheck.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gh.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/zsh-autosuggestions.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libpng.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ncurses.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cmake.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/opus.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpfr.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/highlight.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libvorbis.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/starship.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sox.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/rustup-init.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/clang-format.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libyaml.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libogg.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/circleci.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/readline.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gcc.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/lcov.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/nvm.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/lua.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/htop.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openjdk.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mad.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/flac.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/signal-cli.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/shellcheck.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/bfg.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/isl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl@1.1.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pcre.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/portaudio.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fltk.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/opusfile.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jpeg.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/zsh-syntax-highlighting.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libsndfile.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ruby.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/restic.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libmpc.rb
==> Uninstalling Cask ibm-aspera-connect
==> Uninstalling Cask ibm-aspera-connect
==> Uninstalling artifacts
==> 2 artifacts defined
#<SortedSet: {#<Cask::Artifact::Uninstall:0x00007fd24516ad18 @cask=#<Cask::Cask:0x00007fd2451498c0 @token="ibm-aspera-connect", @sourcefile_path=#<Pathname:/usr/local/Caskroom/ibm-aspera-connect/.metadata/3.10.1.181943/20210105131226.918/Casks/ibm-aspera-connect.rb>, @tap=nil, @block=#<Proc:0x00007fd245149730@/usr/local/Caskroom/ibm-aspera-connect/.metadata/3.10.1.181943/20210105131226.918/Casks/ibm-aspera-connect.rb:1>, @default_config=#<Cask::Config:0x00007fd2451496e0 @explicit={}>, @caskroom_path=#<Pathname:/usr/local/Caskroom/ibm-aspera-connect>, @metadata_master_container_path=#<Pathname:/usr/local/Caskroom/ibm-aspera-connect/.metadata>, @config=#<Cask::Config:0x00007fd24915e6b8 @explicit={}, @default={:languages=>["en-GB", "en-DE", "de-DE", "en"], :appdir=>#<Pathname:/Applications>, :colorpickerdir=>#<Pathname:/Users/chohner/Library/ColorPickers>, :prefpanedir=>#<Pathname:/Users/chohner/Library/PreferencePanes>, :qlplugindir=>#<Pathname:/Users/chohner/Library/QuickLook>, :mdimporterdir=>#<Pathname:/Users/chohner/Library/Spotlight>, :dictionarydir=>#<Pathname:/Users/chohner/Library/Dictionaries>, :fontdir=>#<Pathname:/Users/chohner/Library/Fonts>, :servicedir=>#<Pathname:/Users/chohner/Library/Services>, :input_methoddir=>#<Pathname:/Users/chohner/Library/Input Methods>, :internet_plugindir=>#<Pathname:/Users/chohner/Library/Internet Plug-Ins>, :audio_unit_plugindir=>#<Pathname:/Users/chohner/Library/Audio/Plug-Ins/Components>, :vst_plugindir=>#<Pathname:/Users/chohner/Library/Audio/Plug-Ins/VST>, :vst3_plugindir=>#<Pathname:/Users/chohner/Library/Audio/Plug-Ins/VST3>, :screen_saverdir=>#<Pathname:/Users/chohner/Library/Screen Savers>}, @env={}>, @dsl=#<Cask::DSL:0x00007fd249166c00 @cask=#<Cask::Cask:0x00007fd2451498c0 ...>, @token="ibm-aspera-connect", @version="3.10.1.181943", @sha256=#<Checksum:0x00007fd249166840 @hexdigest="aff234db6ce414044a7932926137a8f451dd6bcc026c1b5f067283d43e1d18c7">, @url=#<URL:0x00007fd249165aa8 @uri=#<URI::HTTPS https://d3gcli72yxqn2z.cloudfront.net/connect_3_10_1_181943_ga/v4/bin/IBMAsperaConnectInstallerOneClick-3.10.1.181943.dmg>, @verified="d3gcli72yxqn2z.cloudfront.net/", @using=nil, @tag=nil, @branch=nil, @revisions=nil, @revision=nil, @trust_cert=nil, @cookies=nil, @referer=nil, @header=nil, @user_agent=:default, @data=nil, @specs={:verified=>"d3gcli72yxqn2z.cloudfront.net/", :user_agent=>:default}, @from_block=false, @caller_location="/usr/local/Caskroom/ibm-aspera-connect/.metadata/3.10.1.181943/20210105131226.918/Casks/ibm-aspera-connect.rb:5:in `block in load'">, @name=["IBM Aspera Connect"], @desc="Facilitate uploads and downloads with an Aspera transfer server", @homepage="https://www.ibm.com/aspera/connect/", @artifacts=#<SortedSet: {...}>, @language_eval=nil, @appcast=#<Cask::DSL::Appcast:0x00007fd249164ea0 @uri=#<URI::HTTPS https://www.ibm.com/aspera/connect/>, @parameters={:must_contain=>"3_10_1_181943"}, @must_contain="3_10_1_181943">>>, @directives={:script=>{:executable=>"~/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh", :args=>["-f"]}, :delete=>"~/Library/Logs/Aspera_Connect", :signal=>[]}>, #<Cask::Artifact::Installer:0x00007fd24516b880 @cask=#<Cask::Cask:0x00007fd2451498c0 @token="ibm-aspera-connect", @sourcefile_path=#<Pathname:/usr/local/Caskroom/ibm-aspera-connect/.metadata/3.10.1.181943/20210105131226.918/Casks/ibm-aspera-connect.rb>, @tap=nil, @block=#<Proc:0x00007fd245149730@/usr/local/Caskroom/ibm-aspera-connect/.metadata/3.10.1.181943/20210105131226.918/Casks/ibm-aspera-connect.rb:1>, @default_config=#<Cask::Config:0x00007fd2451496e0 @explicit={}>, @caskroom_path=#<Pathname:/usr/local/Caskroom/ibm-aspera-connect>, @metadata_master_container_path=#<Pathname:/usr/local/Caskroom/ibm-aspera-connect/.metadata>, @config=#<Cask::Config:0x00007fd24915e6b8 @explicit={}, @default={:languages=>["en-GB", "en-DE", "de-DE", "en"], :appdir=>#<Pathname:/Applications>, :colorpickerdir=>#<Pathname:/Users/chohner/Library/ColorPickers>, :prefpanedir=>#<Pathname:/Users/chohner/Library/PreferencePanes>, :qlplugindir=>#<Pathname:/Users/chohner/Library/QuickLook>, :mdimporterdir=>#<Pathname:/Users/chohner/Library/Spotlight>, :dictionarydir=>#<Pathname:/Users/chohner/Library/Dictionaries>, :fontdir=>#<Pathname:/Users/chohner/Library/Fonts>, :servicedir=>#<Pathname:/Users/chohner/Library/Services>, :input_methoddir=>#<Pathname:/Users/chohner/Library/Input Methods>, :internet_plugindir=>#<Pathname:/Users/chohner/Library/Internet Plug-Ins>, :audio_unit_plugindir=>#<Pathname:/Users/chohner/Library/Audio/Plug-Ins/Components>, :vst_plugindir=>#<Pathname:/Users/chohner/Library/Audio/Plug-Ins/VST>, :vst3_plugindir=>#<Pathname:/Users/chohner/Library/Audio/Plug-Ins/VST3>, :screen_saverdir=>#<Pathname:/Users/chohner/Library/Screen Savers>}, @env={}>, @dsl=#<Cask::DSL:0x00007fd249166c00 @cask=#<Cask::Cask:0x00007fd2451498c0 ...>, @token="ibm-aspera-connect", @version="3.10.1.181943", @sha256=#<Checksum:0x00007fd249166840 @hexdigest="aff234db6ce414044a7932926137a8f451dd6bcc026c1b5f067283d43e1d18c7">, @url=#<URL:0x00007fd249165aa8 @uri=#<URI::HTTPS https://d3gcli72yxqn2z.cloudfront.net/connect_3_10_1_181943_ga/v4/bin/IBMAsperaConnectInstallerOneClick-3.10.1.181943.dmg>, @verified="d3gcli72yxqn2z.cloudfront.net/", @using=nil, @tag=nil, @branch=nil, @revisions=nil, @revision=nil, @trust_cert=nil, @cookies=nil, @referer=nil, @header=nil, @user_agent=:default, @data=nil, @specs={:verified=>"d3gcli72yxqn2z.cloudfront.net/", :user_agent=>:default}, @from_block=false, @caller_location="/usr/local/Caskroom/ibm-aspera-connect/.metadata/3.10.1.181943/20210105131226.918/Casks/ibm-aspera-connect.rb:5:in `block in load'">, @name=["IBM Aspera Connect"], @desc="Facilitate uploads and downloads with an Aspera transfer server", @homepage="https://www.ibm.com/aspera/connect/", @artifacts=#<SortedSet: {...}>, @language_eval=nil, @appcast=#<Cask::DSL::Appcast:0x00007fd249164ea0 @uri=#<URI::HTTPS https://www.ibm.com/aspera/connect/>, @parameters={:must_contain=>"3_10_1_181943"}, @must_contain="3_10_1_181943">>>, @path=#<Pathname:IBM Aspera Connect Installer.app>, @args=[]>}>
==> Uninstalling artifact of class Cask::Artifact::Uninstall
==> Running uninstall script ~/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh
\~/Library/Application\ Support/Aspera/Aspera\ Connect/uninstall_connect.sh -f
Error: Failure while executing; `\~/Library/Application\ Support/Aspera/Aspera\ Connect/uninstall_connect.sh -f` exited with 127. Here's the output:
No such file or directory - ~/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh
/usr/local/Homebrew/Library/Homebrew/system_command.rb:247:in `assert_success!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:63:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:38:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:310:in `uninstall_script'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:76:in `dispatch_uninstall_directive'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:15:in `block in uninstall_phase'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:14:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:14:in `uninstall_phase'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:470:in `block in uninstall_artifacts'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/set.rb:777:in `each'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/set.rb:777:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:467:in `uninstall_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:416:in `uninstall'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:61:in `block in uninstall_casks'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:49:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:49:in `uninstall_casks'
/usr/local/Homebrew/Library/Homebrew/cmd/uninstall.rb:69:in `uninstall'
/usr/local/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'
```

#### Output of `brew doctor --verbose`

```
~ ❯ brew doctor --verbose
==> Cask Environment Variables:
BUNDLE_PATH
CHRUBY_VERSION
GEM_HOME
GEM_PATH
HOMEBREW_CASK_OPTS
LC_ALL
PATH
RBENV_VERSION
RUBYLIB
RUBYOPT
RUBYPATH
SHELL
==> $LOAD_PATHS
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/thor-1.0.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-runtime-stub-0.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-macho-2.5.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-sorbet-0.5.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rspec-2.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rails-2.9.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-performance-1.9.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-progressbar-1.11.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-wait-0.0.9/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-retry-0.6.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-its-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ronn-0.7.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rdiscount-2.2.0.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rack-2.2.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/pry-0.13.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/plist-3.6.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/patchelf-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rainbow-3.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mustache-1.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/method_source-1.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mechanize-2.7.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/webrobots-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ntlm-http-0.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-persistent-4.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-digest_auth-1.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-3.3.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/http-cookie-1.0.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/hpricot-0.8.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/elftools-1.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/domain_name-0.5.20190701/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf-0.1.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf_ext-0.0.7.7/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/diff-lcs-1.4.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/connection_pool-2.2.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/commander-4.5.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/highline-2.0.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/colorize-0.8.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/coderay-1.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov_json_formatter-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/byebug-11.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/bindata-2.4.8/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/activesupport-6.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/zeitwerk-2.4.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tzinfo-2.0.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/i18n-1.8.7/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/concurrent-ruby-1.1.8/lib/concurrent-ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/x86_64-darwin14
/usr/local/Homebrew/Library/Homebrew
/usr/bin/xattr
/usr/bin/swift /usr/local/Homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Homebrew Version
2.7.7-34-g0a46e34
==> macOS
10.15.7
==> SIP
Enabled
/usr/libexec/java_home --xml --failfast
==> Java
N/A
==> Homebrew Cask Staging Location
/usr/local/Caskroom
==> Homebrew Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (3798 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (175 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-fonts (1602 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-drivers (171 casks)
/usr/local/Homebrew/Library/Taps/mtslzr/homebrew-marmaduke-chromium (3 casks)
/usr/bin/xattr
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.


Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /usr/local/Caskroom/miniconda/base/bin/icu-config
  /usr/local/Caskroom/miniconda/base/bin/freetype-config
  /usr/local/Caskroom/miniconda/base/bin/xslt-config
  /usr/local/Caskroom/miniconda/base/bin/libpng16-config
  /usr/local/Caskroom/miniconda/base/bin/libpng-config
  /usr/local/Caskroom/miniconda/base/bin/xml2-config
  /usr/local/Caskroom/miniconda/base/bin/python3-config
  /usr/local/Caskroom/miniconda/base/bin/ncursesw6-config
  /usr/local/Caskroom/miniconda/base/bin/python3.8-config
```

#### Output of `brew tap`

```
~ ❯ brew tap     
                         
homebrew/cask
homebrew/cask-drivers
homebrew/cask-fonts
homebrew/cask-versions
homebrew/core
mtslzr/marmaduke-chromium
vitorgalvao/tiny-scripts
```

**Comment by cho-m** [comment]
It looks like `uninstall script` doesn't do `~` expansion.
Since this is a Shell feature, Ruby needs explicit code to behave similarly.

One thing to try would be see if you can edit Cask and switch `~` with HOME variable.
Maybe like following diff inside `brew edit ibm-aspera-connect`:
```diff
   uninstall script: {
-    executable: "~/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh",
+    executable: "#{ENV["HOME"]}/Library/Application Support/Aspera/Aspera Connect/uninstall_connect.sh",
```

---

On the other hand, `uninstall delete` does have code for this:
- [uninstall_delete](https://github.com/Homebrew/brew/blob/320bccb38d8bbbf8c8c3c4fcd3f8ec34dcc123c5/Library/Homebrew/cask/artifact/abstract_uninstall.rb#L357) runs
  ```ruby
  each_resolved_path(:delete, paths) do |path, resolved_paths|
  ```
- [each_resolved_path](https://github.com/Homebrew/brew/blob/88c9373eb440a136e7be909765ce53607c8ba6b6/Library/Homebrew/cask/artifact/abstract_uninstall.rb#L330) runs
  ```ruby
  resolved_path = resolved_path.expand_path if path.to_s.start_with?("~")
  ```

Probably need one of Cask members/maintainers to comment whether `~` expansion should also be supported inside `uninstall script`.

**Comment by vitorgalvao** [comment]
> Probably need one of Cask members/maintainers to comment whether `~` expansion should also be supported inside `uninstall script`.

Don’t see why not. Care to make a PR for discussion?

**Comment by chohner** [comment]
This isn't awaiting my reply, is it? I don't think I can help out with anything (except happy to test of course)

**Comment by vitorgalvao** [comment]
> This isn't awaiting my reply, is it?

No (but thank you for checking), waiting on @cho-m to decide how to proceed.

**Comment by cho-m** [comment]
@vitorgalvao
I can take a look maybe this weekend to test out possible solutions before creating PR. Since this would be for `Homebrew/brew`, I would need to spend some time checking any unittest requirements.

For implementation, two options I can think of are:

1. Lower impact of just doing "~" expansion inside uninstall script. Probably what I will create PR for. I haven't actually tried it out yet:
```diff
diff --git a/Library/Homebrew/cask/artifact/abstract_uninstall.rb b/Library/Homebrew/cask/artifact/abstract_uninstall.rb
index 3daf5e2ab..6efb2ce6d 100644
--- a/Library/Homebrew/cask/artifact/abstract_uninstall.rb
+++ b/Library/Homebrew/cask/artifact/abstract_uninstall.rb
@@ -296,6 +296,7 @@ module Cask
         ohai "Running uninstall script #{executable}"
         raise CaskInvalidError.new(cask, "#{stanza} :#{directive_name} without :executable.") if executable.nil?

+        executable = File.expand_path(executable) if executable.start_with?("~")
         executable_path = staged_path_join_executable(executable)

         if (executable_path.absolute? && !executable_path.exist?) ||
```

2. Larger impact of modifying `staged_path_join_executable` to use `expand_path` rather than `join` logic. Change will impact install logic and may have unintended impact of expanding all `..`:
```diff
diff --git a/Library/Homebrew/cask/artifact/abstract_artifact.rb b/Library/Homebrew/cask/artifact/abstract_artifact.rb
index 9d6e5d138..d29a5466f 100644
--- a/Library/Homebrew/cask/artifact/abstract_artifact.rb
+++ b/Library/Homebrew/cask/artifact/abstract_artifact.rb
@@ -34,7 +34,7 @@ module Cask
         absolute_path = if path.absolute?
           path
         else
-          cask.staged_path.join(path)
+          path.expand_path(cask.staged_path)
         end

         FileUtils.chmod "+x", absolute_path if absolute_path.exist? && !absolute_path.executable?
```

**Comment by chohner** [comment]
For what its worth, I just ran into this today when trying to reinstall :/

**Comment by cho-m** [comment]
@chohner Sorry, I haven't had a chance to add `~` expansion due to lack of time to figure out Homebrew/brew unittest requirements.

Let me just do easy fix for IBM Aspera Connect cask to use currently supported HOME variable.

**Comment by cho-m** [comment]
@chohner Added PR for simple fix that at least worked on my machine.
Also moved log deletion to `zap trash` along with many other files/folders I found left over on my system.
If you want to do full deletion of all extra data, you will need to use [`--zap` flag](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md).

@vitorgalvao For `~` expansion in `uninstall`, that can be continued separately since it is an extra feature that is not required to close this issue.

**Comment by cho-m** [comment]
Based on review, the plan is to fix issue with tilde expansion support.

A PR has been opened in homebrew/brew. Based on local tests, it should be able to uninstall `ibm-aspera-connect`, assuming the user follows manual install (will fail if user doesn't as script only exists after manually running installer).

**Comment by vitorgalvao** [comment]
Closing in favour of the open PR. Thank you @cho-m!

**Comment by chohner** [comment]
just confirming here that everything works 💯
